src/NServiceBus.Metrics.PerformanceCounters.Tests/PerformanceCountersCacheTests.cs

```diff
@@ -0,0 +1,64 @@
+﻿namespace Tests
+{
+    using System;
+    using ApprovalTests;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class PerformanceCountersCacheTests
+    {
+        [Test]
+        public void Get_the_same_counter_twice_returns_the_same()
+        {
+            var cache = new TestablePerformanceCountersCache();
+
+            var counterName = "CounterName";
+            var instanceName = "InstancenName";
+
+            var firstCounter = cache.Get(new CounterInstanceName(counterName, instanceName));
+            var secondCounter = cache.Get(new CounterInstanceName(counterName, instanceName));
+
+            Assert.NotNull(firstCounter);
```

Comment:
I asked because if it _doesn't_ return true when they are both null, we can remove the `NotNull` checks completely.